src/solutions/ode_solutions.jl

```diff
@@ -15,7 +15,40 @@ struct ODESolution{T,N,uType,uType2,DType,tType,rateType,P,A,IType,DE} <: Abstra
   destats::DE
   retcode::Symbol
 end
-(sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left) = sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+
+function (sol::ODESolution)(t,deriv::Type=Val{0};idxs=nothing,continuity=:left)
+  if deriv != Val{0}
+    error("Not Implemented")
+  end
+
+  if idxs === nothing
+    if t isa Real
+      sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+    else
+      augment(sol.interp(t,idxs,deriv,sol.prob.p,continuity), sol)
+    end
+  elseif idxs isa Vector{<:Integer} || idxs isa Integer
+      sol.interp(t,idxs,deriv,sol.prob.p,continuity)
+  elseif issymbollike(idxs)
+    if t isa Real
+      interp_sol = augment(sol.interp([t],nothing,deriv,sol.prob.p,continuity), sol)
+      interp_sol[idxs][1]
+    else
+      interp_sol = augment(sol.interp(t,nothing,deriv,sol.prob.p,continuity), sol)
+      @info interp_sol[idxs]
+      DiffEqArray(interp_sol[idxs], t)
+    end
+  else
+    if t isa Real
+      interp_sol = augment(sol.interp([t],nothing,deriv,sol.prob.p,continuity), sol)
```

Comment:
Same here, can't we use

```suggestion
      interp_sol = augment(sol.interp(t,nothing,deriv,sol.prob.p,continuity), sol)
```